### ISSUE TITLE
Korean and CJK String send Problem.

### ISSUE THREAD

**Issue by sesangsokuro** [marvin-we/steem-java-api-wrapper#33]
when i send, "안녕하세요" ( it means 'hi or hello' ) in korean, steemj print error.

Unrecognized field "active" but it was correct, why? when i change the comment's body to english ('hi').

It works! I think. it was cjk character problem or utf-8 encoding problem!

--nrecognized field "active" (class eu.bittrade.libs.steem.api.wrapper.models.error.SteemData), not marked as ignorable (8 known properties: "call.method", "posting", "what", "call.params", "type", "name", "id", "api"])
 at [Source: {"id":11,"error":{"code":1,"message":"3030000 tx_missing_posting_auth: missing required posting authority\nMissing Posting Authority sesangsokuro\n    {\"id\":\"sesangsokuro\",\"posting\":{\"weight_threshold\":1,\"account_auths\":[],\"key_auths\":
----------------------something my key ----------------
th_a  transaction.cpp:129 verify_authority\n\n    {\"ops\":[[\"comment\",{\"parent_author\":\"jin.lee\",\"parent_permlink\":\"6smnbc\",\"author\":\"sesangsokuro\",\"permlink\":\"re-6smnbc-20170625t180512535z\",\"title\":\"\",\"body\":\"안녕하세요\",\"json_metadata\":\"{\\\"tags\\\":[\\\"kr-newbie\\\"],\\\"app\\\":\\\"steemup/0.1\\\"}\"}]],\"sigs\":

...

line: 1, column: 4104] (through reference chain: eu.bittrade.libs.steem.api.wrapper.models.error.SteemError["error"]->eu.bittrade.libs.steem.api.wrapper.models.error.SteemErrorDetails["data"]->eu.bittrade.libs.steem.api.wrapper.models.error.SteemErrorData["stack"]->java.lang.Object[][0]->eu.bittrade.libs.steem.api.wrapper.models.error.SteemStack["data"]->eu.bittrade.libs.steem.api.wrapper.models.error.SteemData["active"])
	at com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:62)
	at com.fasterxml.jackson.databind.DeserializationContext.handleUnknownProperty(DeserializationContext.java:834)
	at com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:1093)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty(BeanDeserializerBase.java:1478)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownVanilla(BeanDeserializerBase.java:1456)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:282)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:140)
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:504)
	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:111)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:276)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:140)
	at com.fasterxml.jackson.databind.deser.std.ObjectArrayDeserializer.deserialize(ObjectArrayDeserializer.java:196)
	at com.fasterxml.jackson.databind.deser.std.ObjectArrayDeserializer.deserialize(ObjectArrayDeserializer.java:20)
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:504)
	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:111)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:276)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:140)
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:504)
	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:111)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:276)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:140)
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:504)
	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:111)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:276)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:140)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:3798)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2842)
	at eu.bittrade.libs.steem.api.wrapper.communication.CommunicationHandler.performRequest(CommunicationHandler.java:161)
	... 3 more

I can't handle this. help!

**Comment by marvin-we** [comment]
Hay @sesangsokuro,

thank you once again for your feedback and helping to improve this project. I think we have two issues here:
1. The error parsing/handling is quite bad. I am aware of this and now created #34 to keep this issue in mind.
2. The root problem of this issue. The first good thing is that I am able to reproduce it. SteemJ supports multiple encodings in theory. You can configure the charset like this:

```Java
SteemJConfig.getInstance().setEncodingCharset(Charset.forName("UTF-8"));
```

But actually if you do so SteemJ dies with another error. I'll try to investigate on that and come back to you with a fix or a working solution.

Best regards!

**Comment by marvin-we** [comment]
Hallo again @sesangsokuro,

you had a bad timing here :( I've just created the version 0.3.0 and your problem requires a code fix. If you need this fix and do not want to wait for the next release you could have a look at the updated [integration test](https://github.com/marvin-we/steem-java-api-wrapper/blob/master/src/test/java/eu/bittrade/libs/steemj/base/models/operations/CommentOperationIT.java).

Using the latest revision your problem should be solved by setting the charset to UTF-8 as shown in the post above.

Best regards!

**Comment by sesangsokuro** [comment]
After I leave work at home, I will try this method.

I think I'm asking too many questions. Would you mind?

But I'll add another questions. haha! :D

thanks you marvin-we, with your fast feed back

1. Do I need the code below for comment ?
  ( When I write before line like this )
    myConfig.setPrivateKey( // Privatekey.Post, "ST5~~~~~~~~~~~~~~keystringABDE" ); )


  Authority posting = new Authority();
  posting.setAccountAuths(new HashMap<>());
  Map<PublicKey, Integer> postingKeyAuth = new HashMap<>();
  postingKeyAuth.put(new PublicKey("STM8CemMDjdUWSV5wKotEimhK6c4dY7p2PdzC2qM1HpAP8aLtZfE7"), 1);
  posting.setKeyAuths(postingKeyAuth);
  posting.setWeightThreshold(1);

  // Authority error seems to say something about this. ( But why did it work when it was run in English? )

2. 

how to crawl a posting content over a 100 about specific tags.

// success. But i wanna crawl before this 100 blog posting. 
// if it return content's id was 1000~1100
// I want to get before 1000 post. ( ex. 900 901 ~ 500... etc )
// how do i get it?

List<Content> contents = steemApiWrapper.getDiscussionsBy("kr", 100, DiscussionSortType.SORT_BY_BLOG);


// I want to make a magazine like a piece of the best newbie article for a week for a certain tag.

**Comment by marvin-we** [comment]
Hay @sesangsokuro,

thanks for your answer. To answer your questions:

# 1. Question
You need

```Java
myConfig.setPrivateKey( // Privatekey.Post, "ST5~~~~~~~~~~~~~~keystringABDE" ); )
````

For the comment operation and not more =) It only needs your private posting key to sign the operation.

The following part:
```Java
Authority posting = new Authority();
posting.setAccountAuths(new HashMap<>());
Map<PublicKey, Integer> postingKeyAuth = new HashMap<>();
postingKeyAuth.put(new PublicKey("STM8CemMDjdUWSV5wKotEimhK6c4dY7p2PdzC2qM1HpAP8aLtZfE7"), 1);
posting.setKeyAuths(postingKeyAuth);
posting.setWeightThreshold(1);

// Authority error seems to say something about this. ( But why did it work when it was run in English? )
```
Is, for example, needed for the AccountCreateOperation to define the new keys of the account, but not for other methods.

# 2 Question
Here is how the original Steem code looks like:
```C++
struct discussion_query {
   void validate()const{
      FC_ASSERT( filter_tags.find(tag) == filter_tags.end() );
      FC_ASSERT( limit <= 100 );
   }
```

So as you can see it is sadly not possible to pull over 100 entries using the get_discussion_method :( Could you create a separate issue for this? I would like to investigate a little bit more and try to find out if its really not possible.

A workaround could be to analyse all operations in all blocks (from getBlock(1) to getBlock(13342345)). Each block contains all transactions / operations that have been processed (getBlock(1).getTransactions(0).getOperations(1...)). If this operation is an instance of CommentOperation you can have a closer look at it and inpsect the used tags (which should be present in the jsonMetadata field). I hope that was understandable. As an example you can have [a look how my latest SteemJ update post looks like on Steemd](https://steemd.com/tx/407773bbdf75a909fdba69089d29c1955d6c9ab4).

3. Request
I think you already have a pretty good knowledge about the SteemJ and Steem-Api in general, so I just have to ask you now: Would you like to help me with some Wiki articles? As you know SteemJ has a huge lack of documentation and I guess this makes it quite unsable for newcommers and I am not sure if I have enough time to solve this in the near future =(. I would really appreciate it.  

Best regards!

**Comment by sesangsokuro** [comment]
Sorry for the delay. 

Writing related articles on the wiki is something I would like to help. But there are two things that I can do as a side job and I do not think I can help you fully. However, I can add an entry for the blocked parts during coding. Will that even help with this project?

p.s.

The Korean issue has been fixed in version 0.31. its thanks to you. happy.
But I think the Maven repository has the latest 0.26, can not update it? I used the source of the feathers and installed it in Maven.

**Comment by marvin-we** [comment]
Everything that helps new people to start with SteemJ without troubles helps for sure! Thank you for your effort :)

The latest version in the Maven Central Repo is 0.3.0 now. (take a look [here](https://mvnrepository.com/artifact/eu.bittrade.libs/steem-api-wrapper/0.3.0)). When I create an Update post I make sure to push the new version to the central repo, but it takes up to 24 hours until its synced - Maybe thats why it wasn't available for you :/
